pyhole/plugins/ops.py

```diff
@@ -29,59 +30,245 @@ def __init__(self, session):
         self.subdomain = pagerduty.get("subdomain")
-        self.key = pagerduty.get("key")
+        self.api_key = pagerduty.get("api_key")
+        self.integration_key = pagerduty.get("integration_key")
 
-        self.level = "GREEN"
-
-    @plugin.hook_add_command("note")
-    @utils.require_params
-    @utils.spawn
-    def note(self, message, params=None, **kwargs):
-        """Manage notes during an outage (ex: .note <message>)."""
-        return
+        self.api_headers = {
+            "Authorization": "Token token=%s" % self.api_key,
+            "Content-Type": "application/json"
+        }
+        self.user_cache = None
+        self.reset_current_incident()
 
     @plugin.hook_add_command("oncall")
     @utils.spawn
     def oncall(self, message, params=None, **kwargs):
         """Show who is on call (ex: .oncall [<group>])."""
         url = "%s/api/v1/escalation_policies/on_call" % self.subdomain
-        headers = {
-            "Authorization": "Token token=%s" % self.key,
-            "Content-Type": "application/json"
-        }
+        request = utils.get_url(
+            url,
+            headers=self.api_headers,
+            params={"query": params})
 
-        response_json = utils.fetch_url(url, headers=headers,
-                                        params={"query": params}).json()
+        if request.status_code == requests.codes.ok:
+            response_json = request.json()
+            for policy in response_json["escalation_policies"]:
+                message.dispatch(policy["name"])
 
-        for policy in response_json["escalation_policies"]:
-            message.dispatch(policy["name"])
+                for level in policy["on_call"]:
+                    message.dispatch("- Level %s: %s <%s>" % (
+                                     level["level"],
+                                     level["user"]["name"],
+                                     level["user"]["email"]))
+        else:
+            message.dispatch("Could not get on call list: status code %d" %
+                             request.status_code)
 
-            for level in policy["on_call"]:
-                message.dispatch("- Level %s: %s <%s>" % (
-                    level["level"],
-                    level["user"]["name"],
-                    level["user"]["email"]))
+    @plugin.hook_add_command("create_incident")
+    @utils.require_params
+    @utils.spawn
+    def create_incident(self, message, params=None, **kwargs):
+        """Create a PagerDuty incident (ex: .create_incident <message>)."""
+        url = "https://events.pagerduty.com/generic/2010-04-15/create_event.json"
+        data = {"service_key": self.integration_key,
+                "event_type": "trigger",
+                "description": params,
+                "incident_key": utils.datetime_now_string(),
+                "details": {"triggered by": message.source}}
 
-            time.sleep(2)
+        request = utils.post_url(url, data=json.dumps(data))
```

Comment:
POSTs in requests supports the `json` kwarg -- could we get by w/o the json import above, and pass the dict as `json=data` here?